components/invitation_modal/invite_view.tsx

```diff
@@ -0,0 +1,339 @@
+import InviteAs, {As} from './invite_as';
+import './invite_view.scss';
+
+export const defaultInviteState: InviteState = deepFreeze({
+    as: As.MEMBER,
+    customMessage: defaultCustomMessage,
+    sending: false,
+    inviteChannels: defaultInviteChannels,
+    usersEmails: [],
+    usersEmailsSearch: '',
+});
+
+export type InviteState = {
+    customMessage: CustomMessageProps;
+    as: As;
+    sending: boolean;
+    inviteChannels: InviteChannels;
+    usersEmails: Array<UserProfile | string>;
+    usersEmailsSearch: string;
+};
+
+export type Props = InviteState & {
+    setInviteAs: (as: As) => void;
+    invite: () => void;
+    onChannelsChange: (channels: Channel[]) => void;
+    onChannelsInputChange: (channelsInputValue: string) => void;
+    onClose: () => void;
+    currentTeam: Team;
+    currentChannelName: string;
+    inviteToTeamTreatment: InviteToTeamTreatments;
+    setCustomMessage: (message: string) => void;
+    toggleCustomMessage: () => void;
+    channelsLoader: (value: string, callback?: (channels: Channel[]) => void) => Promise<Channel[]>;
+    regenerateTeamInviteId: (teamId: string) => void;
+    isAdmin: boolean;
+    usersLoader: (value: string, callback: (users: UserProfile[]) => void) => Promise<UserProfile[]> | undefined;
+    onChangeUsersEmails: (usersEmails: Array<UserProfile | string>) => void;
+    isCloud: boolean;
+    subscriptionStats?: SubscriptionStats | null;
+    cloudUserLimit: string;
+    emailInvitationsEnabled: boolean;
+    onUsersInputChange: (usersEmailsSearch: string) => void;
+    headerClass: string;
+    footerClass: string;
+    canInviteGuests: boolean;
+    canAddUsers: boolean;
+}
+
+function clearMinHeight() {
+    const modalBody = document.querySelector('.modal-body') as HTMLDivElement;
+    if (!modalBody) {
+        return;
+    }
+    if (modalBody.style.minHeight) {
+        modalBody.style.minHeight = '';
+    }
+}
+
+const FOOTER_HEIGHT = 63;
+
+const ensureInvitesVisible = debounce((options: UserProfile[]) => {
+    if (options.length < 2) {
+        clearMinHeight();
+        return;
+    }
+    setTimeout(() => {
+        const modalBody = document.querySelector('.modal-body') as HTMLDivElement;
+        if (!modalBody) {
+            return;
+        }
+
+        const bodyRect = modalBody.getBoundingClientRect();
+        const menu = document.querySelector('.users-emails-input__menu');
+        if (!menu) {
+            return;
+        }
+
+        const menuRect = menu.getBoundingClientRect();
+
+        if (menuRect.bottom > bodyRect.bottom) {
+            let betterHeight = menuRect.bottom - bodyRect.top;
+            if ((bodyRect.top + betterHeight + FOOTER_HEIGHT) > window.innerHeight) {
+                betterHeight = window.innerHeight - bodyRect.top - FOOTER_HEIGHT;
+            }
+            modalBody.style.minHeight = `${betterHeight}px`;
+        }
+    }, 125);
```

Comment:
A guess/compromise that for most devices, this will be enough time to render given that the data has already been fetched from the network. In the worst case, we sometimes fail to make this tall enough on slower devices. As an enhancement, we could build in retry logic, but wanted to avoid that if we can.

Reply:
nit: other option would be show a loading spinner and once data is loaded (check in componentUpdate lifecycle), show the modal content data. Is just me not being a fan of using timeOuts :p